resolwe/flow/models/utils/duplicate.py

```diff
@@ -0,0 +1,482 @@
+"""Resolwe models duplicate utils."""
+from copy import deepcopy
+
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, models, transaction
+from django.utils import timezone
+from guardian.models import GroupObjectPermission, UserObjectPermission
+
+from resolwe.flow.utils import iterate_fields
+
+
+def _check_permissions(objects, permission, user):
+    """Check that user has permission on all given objects.
+
+    :param objects: A list of objects to be checked.
+    :type objects: list of `~django.db.models.Model`
+
+    :param str permission: Codename of the required permission.
+
+    :param contributor: A Django user that should have the permission.
+    :type contributor: `~django.contrib.auth.models.User`
+
```

Comment:
Maybe add something similar to the line bellow to the documentation.

`:raises ValidationError when user does not have all the required permissions.`